coral/src/app/features/topics/request/TopicRequest.test.tsx

```diff
@@ -489,6 +494,33 @@ describe("<TopicRequest />", () => {
 
         expect(errorMessage).not.toBeVisible();
       });
+
+      it("does not validate the prefix if a regex should be applied", async () => {
+        const expectedErrorMsg = 'Topic name must start with "test-".';
+        const select = await screen.findByRole("combobox", {
+          name: "Environment *",
+        });
+
+        await user.selectOptions(select, "EnvWithTopicRegex");
+
+        const topicNameInput = screen.getByLabelText(/Topic name/);
+        await user.type(topicNameInput, "foobar{tab}");
+
+        try {
```

Comment:
This looks a bit weird, but it's a good way I found to make sure that the error does not appear. Since it's rendered async, we can't use a `queryBy` selector, since that will look for the element instantly and not find it -> but the element can appear after, so it's a false-green test.

Reply:
Would artificial wait before the check also work here? While not as robust, it could help reduce adding logic to tests.

```
waitFor(() => {
    expect(screen.getByLabelText(/Topic name/)).not.toHaveErrorMessage(expectedErrorMsg)
  }, 300) // Propably more than enough
```

Reply:
The timeout for waitFor is by default already 1000ms at least :D but the callback must return an error to trigger a retry, so in our case it would return true instantly because the message is not there in that moment (even if it could be in the DOM later). 

I checked a potential solution with adding an artificial timeout, but decided against it because I've a high dislike of test validation relying on timeouts. For example, when our tests on the CI pipelines get slower, a test could be green even if functionality breaks, when the error message takes longer then the 300ms to appear in the test DOM. 

With that solution (and I'll admit it looks over complicated) I could get the most reliable feedback -> we know that `await screen.findByText(expectedErrorMsg)` does return an error message if there is one rendered, because it's used in the scenario above too. So if this throws an error, and this error matches the requirements (Object with name "TestingLibraryElementError") we know this is the error that we're expecting.